overlord/devicestate/devicemgr.go

```diff
@@ -1877,52 +1877,19 @@ func (m *DeviceManager) Systems() ([]*System, error) {
 // including the model assertion, gadget details and encryption info
 // for the given system label.
 func (m *DeviceManager) SystemAndGadgetAndEncryptionInfo(wantedSystemLabel string) (*System, *gadget.Info, *EncryptionSupportInfo, error) {
-	if m.isClassicBoot {
-		return nil, nil, nil, fmt.Errorf("cannot get model and gadget information on a classic boot system")
-	}
-	// get current system as input for loadSeedAndSystem()
-	systemMode := m.SystemMode(SysAny)
-	currentSys, _ := currentSystemForMode(m.state, systemMode)
+	// TODO check that the system is not a classic boot one when the
+	// installer is not anymore.
 
-	s, sys, err := loadSeedAndSystem(wantedSystemLabel, currentSys)
+	// System information
+	sys, snapInfos, seedSnaps, err := m.loadSystemAndEssentialSnaps(wantedSystemLabel, []snap.Type{snap.TypeKernel, snap.TypeGadget})
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
-	// 2. get the gadget volumes for the given system-label
-	perf := &timings.Timings{}
-	if err := s.LoadEssentialMeta([]snap.Type{snap.TypeKernel, snap.TypeGadget}, perf); err != nil {
-		return nil, nil, nil, fmt.Errorf("cannot load gadget snap metadata: %v", err)
-	}
-	// EssentialSnaps is always ordered, see asserts.Model.EssentialSnaps:
-	// "snapd, kernel, boot base, gadget." and snaps not loaded above
-	// like "snapd" will be skipped and not part of the EssentialSnaps list
-	//
-	// kernel info needed to check encryptionSupport
-	kernelSnap := s.EssentialSnaps()[0]
-	if kernelSnap.Path == "" {
-		return nil, nil, nil, fmt.Errorf("internal error: cannot get kernel snap path")
-	}
-	snapf, err := snapfile.Open(kernelSnap.Path)
-	if err != nil {
-		return nil, nil, nil, fmt.Errorf("cannot open kernel snap: %v", err)
-	}
-	kernelInfo, err := snap.ReadInfoFromSnapFile(snapf, kernelSnap.SideInfo)
-	if err != nil {
-		return nil, nil, nil, err
-	}
-	if kernelInfo.SnapType != snap.TypeKernel {
-		return nil, nil, nil, fmt.Errorf("cannot use kernel info, expected kernel but got: %v", kernelInfo.SnapType)
-	}
-
-	// gadget info needed to check encryptionSupport and gadget contraints
-	gadgetSnap := s.EssentialSnaps()[1]
-	if gadgetSnap.Path == "" {
-		return nil, nil, nil, fmt.Errorf("internal error: cannot get gadget snap path")
-	}
-	snapf, err = snapfile.Open(gadgetSnap.Path)
+	// Gadget information
+	snapf, err := snapfile.Open(seedSnaps[snap.TypeGadget].Path)
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("cannot open gadget snap: %v", err)
+		return nil, nil, nil, fmt.Errorf("cannot open gadget.yaml: %v", err)
```

Comment:
the error was right before, as this open the snap not the yaml

Reply:
Fixed